### ISSUE TITLE
How to refresh detection?

### ISSUE THREAD

**Issue by rsalmei** [aclap-dev/vdhcoapp#225]
After downloading something from a streaming website, the original content keeps playing, of course. And I can't download a new portion of it without first refreshing the page...
The detected video is gone as soon as we press download, changed into a downloading handle, and when we click stop there are two cases:

- with "Automatically hide downloaded media", it simply vanishes from screen, making the popup empty (but the original video keeps playing).
- without it, the popup only shows a handle to **play** that newly downloaded video, which I find useless.

In any case, I can't download it again. How to make the **download** handle reappear?
Is it possible for it to automatically reappear in case the video is still streaming? If not, how about a Refresh button?
By the way, a Refresh button would nicely complement the Clean and Clean All buttons, one undoing the other.
Thanks!

**Comment by paulrouget** [comment]
> How to make the download handle reappear?

On the left of the Play button, there's a retry button:

<img width="332" alt="Screenshot 2024-05-24 at 09 40 19" src="https://github.com/aclap-dev/vdhcoapp/assets/373579/dad95e73-0eef-4711-97b8-c0ce4deab613">

**Comment by rsalmei** [comment]
Thanks, but unfortunately, it is not the same behavior.
By clicking Retry, the entire frame won't hide again, they will simply stay there on the list even after changing the selected tab. And, after only two or three plus their downloaded ones, the list quickly starts to pile up and become unmanageable. I can barely find the newly detected ones on a new tab anymore.

**Comment by rsalmei** [comment]
When a download is finished, all it would take is to auto-refresh the list of detected videos.

Or, just allow us to show a Refresh button in it:
<img width="337" alt="image" src="https://github.com/aclap-dev/vdhcoapp/assets/6652853/990fc182-5e2f-4aaa-bfed-1e05b20c27dc">

We can already optionally show Clean buttons, which I don't particularly appreciate as I prefer auto-hiding to keep it clean, but a Refresh button is surprisingly missing:
<img width="273" alt="image" src="https://github.com/aclap-dev/vdhcoapp/assets/6652853/773638be-93fc-433f-a284-c5e7effc5ee2">

**Comment by rsalmei** [comment]
Yes, it does, but it's a much more expensive action. It would be much faster if only the addon could do a refresh instead of the whole page, which also has to reconnect chat and video.
I've been doing this, but I don't know why, after a few times it makes my Firefox stop connecting altogether.

**Comment by paulrouget** [comment]
Ah, ok, I totally understand.

That's a totally valid use case. And actually… I guess that's how most people want this to work.

Let me think of something…

**Comment by rsalmei** [comment]
Hi @paulrouget.

Actually, not. I just tested on twitch, on a live stream, and it doesn't work.

For example, I downloaded a portion of a live stream here: https://www.twitch.tv/alanzoka
It generated `alanzoka - Twitch.mp4`, and now the popup is empty:
<img width="459" alt="image" src="https://github.com/aclap-dev/vdhcoapp/assets/6652853/651f4af4-c2e2-4680-8f6e-96f4e076a5d8">
So, to grab another portion, I have to refresh the whole page.

If I uncheck "Automatically hide downloaded media", it does show the previous video where I can click re-download, but with a BIG caveat:
<img width="347" alt="image" src="https://github.com/aclap-dev/vdhcoapp/assets/6652853/e9814f11-2638-47fd-b732-1d0d96a283cb">

If I change tabs now (I have to use several tabs at the same time), the above previous video SHOWS THERE TOO:
<img width="448" alt="image" src="https://github.com/aclap-dev/vdhcoapp/assets/6652853/f2f10c70-78c3-4fdf-aebd-6dbad088d248">
And it quickly becomes overwhelming and unmanageable:
<img width="347" alt="image" src="https://github.com/aclap-dev/vdhcoapp/assets/6652853/19f85aaa-2dd8-4441-beb2-ae897a1458d9">

And what if I click re-download now?? Yikes, it is even worse:
<img width="345" alt="image" src="https://github.com/aclap-dev/vdhcoapp/assets/6652853/240863fe-4089-4c80-a56c-79a2075befcc">
1. the previously downloaded video doesn't go away,
2. a new entry is added to download it again
3. and this entry shows in every single tab I select...

I hope that helps to understand the problem.

**Comment by paulrouget** [comment]
> the previously downloaded video doesn't go away,

You can always click on the "x", no?

> a new entry is added to download it again

Yes.

> and these entries show in every single tab I select...

Yes, I know about this. It's actually necessary, but I think I can work around that issue.

I'll work on improving the Twitch integration. I understand it's not great at the moment.

**Comment by rsalmei** [comment]
I can click on x, sure, but there may be several downloads running at the same time, so it's not that easy to find it. And, I already click STOP, there shouldn't be another required click.

That's why automatically cleaning is the best option for me. This keeps tabs completely isolated from each other.

All that is needed for me is a simpler way to show the video again on the pop up. It is streaming after all. If not automatically, at least a button for doing it manually.
Thank you!

**Comment by paulrouget** [comment]
Ok, I get you! Thank you for the details and all.

I want to improve the unknown-length streaming situation.
